src/repository/index.ts

```diff
@@ -476,6 +507,63 @@ Message: ${err.message}`,
     return tags.map((tag) => `${tag.name}:${tag.value}`);
   }
 
+  private async switchToPolling() {
+    if (this.mode.type === 'polling') {
+      return;
+    }
+
+    this.emit(UnleashEvents.Mode, { from: 'streaming', to: 'polling' });
+
+    if (this.eventSource) {
+      this.eventSource.close();
+      this.eventSource = undefined;
+    }
+
+    this.mode = { type: 'polling', format: 'full' };
```

Comment:
Hard-coding `this.mode = { type: 'polling', format: 'full' }` is bad because it ignores an existing `delta` polling configuration, so carrying over the prior `format` or honoring initial config would prevent unexpected behavior.

Reply:
Ok, just so we do not forget it, as it is ticking bomb for future.